neurogym/wrappers/monitor.py

```diff
@@ -461,3 +496,120 @@ def plot_training_history(
             logger.info(f"Figure saved to {save_path}")
 
         return fig
+
+    def record_activations(
+        self,
+        layer: nn.Module,
+        name: str | None = None,
+        steps: int | None = None,
+    ) -> ActivationMonitor:
+        """Record the output activations of a layer over a trial.
+
+        Args:
+            layer: The layer whose activations are being monitored.
+            name: The name to use for the activation monitor.
+                This can be useful for retrieving activation monitors
+                at a later stage.
+            steps: The steps to record for.
+                This could be less than the total number of steps in a trial.
+
+        Returns:
+            An ActivationMonitor instance.
+        """
+        # Ensure that we have a name for this monitor
+        if name is None:
+            name = layer.__class__.__name__
+
+        # The total number of steps to record during each trial.
+        if steps is None:
+            steps = int(self.tmax / self.dt)
+
```

Comment:
What happens if the number given is larger than the total? Does it make sense to add something like this to explicitly cap it?

```suggestion
        # The total number of steps to record during each trial.
        total_steps = int(self.tmax / self.dt)
        steps = total_steps if steps is None else min(steps, total_steps)
```